### PR TITLE
P10 enablement for SAP

### PR DESCRIPTION
**Pull request by ChaseRun**
- Add ability to deploy SAP instance on E1080
 - Updated SAP resource tests to specify SysType
 - Implemented updated ibmpisession
 - Minor updates to variable names based on updated Power API

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
=== RUN   TestAccIBMPISAPInstance
--- PASS: TestAccIBMPISAPInstance (1436.03s)
PASS
```

### CODE REVIEW

File: ibm/service/power/resource_ibm_pi_instance.go [hunk 1]
```diff
@@ -259,16 +259,15 @@ func ResourceIBMPIInstance() *schema.Resource {
...
-				Type:          schema.TypeString,
-				Optional:      true,
-				Computed:      true,
-				ValidateFunc:  validate.ValidateAllowedStringValues([]string{"s922", "e880", "e980"}),
-				ConflictsWith: []string{PISAPInstanceProfileID},
-				Description:   "PI Instance system type",
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validate.ValidateAllowedStringValues([]string{"s922", "e880", "e980", "e1080"}),
```
Comment (yussufsh) [inline ibm/service/power/resource_ibm_pi_instance.go:269]: IMO we should remove the validate check as more system types are allowed for pvm instance such as `s1022` which is not possible to use via Terraform today.
You could have a dynamic check specific for SAP create func to allow selected system types.

**Review by yussufsh — CHANGES_REQUESTED** [review]

File: website/docs/r/pi_instance.html.markdown [hunk 2]
```diff
@@ -88,8 +88,8 @@ Review the argument references that you can specify for your resource.
 - `pi_storage_pool_affinity` - (Optional, Bool) Indicates if all volumes attached to the server must reside in the same storage pool. The default value is `true`. To attach data volumes from a different storage pool (mixed storage) set to `false` and use `pi_volume_attach` resource. Once set to `false`, cannot be set back to `true` unless all volumes attached reside in the same storage type and pool.
 - `pi_storage_type` - (Optional, String) - Storage type for server deployment. Only valid when you deploy one of the IBM supplied stock images. Storage type for a custom image (an imported image or an image that is created from a VM capture) defaults to the storage type the image was created in
 - `pi_storage_connection` - (Optional, String) - Storage Connectivity Group (SCG) for server deployment. Only supported value is `vSCSI`.
-- `pi_sys_type` - (Optional, String) The type of system on which to create the VM (s922/e880/e980).
-  - Required when not creating SAP instances. Conflicts with `pi_sap_profile_id`.
+- `pi_sys_type` - (Optional, String) The type of system on which to create the VM (s922/e880/e980/e1080).
```
[the remark this answers is not in the document]
Comment (rmoralesjr) [reply website/docs/r/pi_instance.html.markdown:91]: FYI, I have not seen the s1022 in the service broker swagger.yaml in the SAPCreate API

[the remark this answers is not in the document]
Comment (yussufsh) [reply website/docs/r/pi_instance.html.markdown:91]: I meant adding it for pvm create as this attribute is shared by both.

File: ibm/service/power/resource_ibm_pi_instance.go (hunk 1 above)
Comment (ChaseRun) [reply ibm/service/power/resource_ibm_pi_instance.go:269]: I'll just remove the validate check for this and we can let the service broker check for the correct SAP sysType

File: website/docs/r/pi_instance.html.markdown (hunk 2 above)
[the remark this answers is not in the document]
Comment (ChaseRun) [reply website/docs/r/pi_instance.html.markdown]: fixed

**Pull request merged**